server/upload.py

```diff
@@ -53,15 +53,20 @@ def __init__(self, zip_file_path):
         # TODO: check for "7z" command
 
     def __enter__(self):
+        # Create temporary directory
+        self.tempDirManager = TempDir()
+        tempDir = self.tempDirManager.__enter__()
```

Comment:
Is there a reason that you didn't make `tempDir` also an instance member, so we don't have to pass it through to every function?

Reply:
I guess that's better here. I'll update this branch.